```
      em++ -fno-strict-overflow -Wsign-compare -Wunreachable-code -DNDEBUG -g -O3 -Wall -O2 -g0 -fPIC -fwasm-exceptions -sSUPPORT_LONGJMP -DPY_CALL_TRAMPOLINE -I$PREFIX -I$BUILD_PREFIX/venv/lib/python3.13/site-packages/sage/cpython -I$BUILD_PREFIX/venv/lib/python3.13/site-packages/cysignals -I$BUILD_PREFIX/venv/lib/python3.13/site-packages -I$BUILD_PREFIX/venv/lib/python3.13/site-packages/numpy/_core/include -I$PREFIX/include/python3.13 -I$PREFIX/include/python3.13 -c sage/libs/sirocco.cpp -o build/temp.unknown-wasm32-cpython-313/sage/libs/sirocco.o
      sage/libs/sirocco.cpp:6437:17: error: use of undeclared identifier 'homotopyPath_mp'
       6437 |   __pyx_v_rop = homotopyPath_mp(__pyx_v_deg, __pyx_v_cvalues, __pyx_v_y0R, __pyx_v_y0I, __pyx_v_prec);
            |                 ^~~~~~~~~~~~~~~
      sage/libs/sirocco.cpp:7146:17: error: use of undeclared identifier 'homotopyPath_mp_comps'
       7146 |   __pyx_v_rop = homotopyPath_mp_comps(__pyx_v_deg, __pyx_v_cvalues, __pyx_v_y0R, __pyx_v_y0I, __pyx_v_prec, ((int)__pyx_t_1), __pyx_v_cotherdegs, __pyx_v_cothercoefs);
            |                 ^~~~~~~~~~~~~~~~~~~~~
      sage/libs/sirocco.cpp:8394:17: error: use of undeclared identifier 'homotopyPath_comps'
       8394 |   __pyx_v_rop = homotopyPath_comps(__pyx_v_deg, __pyx_v_c_values, __pyx_v_y0R, __pyx_v_y0I, ((int)__pyx_t_1), __pyx_v_c_otherdegrees, __pyx_v_c_othercoefs);
            |                 ^~~~~~~~~~~~~~~~~~
      3 errors generated.
      error: command '$BUILD_PREFIX/bin/em++' failed with exit code 1
```

--- a/recipes/recipes_emscripten/passagemath-sirocco/recipe.yaml
+++ b/recipes/recipes_emscripten/passagemath-sirocco/recipe.yaml
@@ -31,6 +31,7 @@ requirements:
   host:
   - python
   - gmp
+  - mpfr
   - sirocco
   - numpy
   - passagemath-modules

--- a/recipes/recipes_emscripten/passagemath-sirocco/recipe.yaml
+++ b/recipes/recipes_emscripten/passagemath-sirocco/recipe.yaml
@@ -1,0 +1,68 @@
+context:
+  version: 10.8.6
+
+package:
+  name: passagemath-sirocco
+  version: ${{ version }}
+
+source:
+- url: https://pypi.org/packages/source/p/passagemath-sirocco/passagemath_sirocco-${{ version }}.tar.gz
+  sha256: e950a7504637560d589c9dcca3bd1cfb1abab533134d85de54126294ababa829
+
+build:
+  script: ${{ PYTHON }} -m pip install .
+  number: 0
+
+requirements:
+  build:
+  - python
+  - crossenv >=1.2
+  - cross-python_emscripten-wasm32
+  - ${{ compiler("c") }}
+  - ${{ compiler("cxx") }}
+  - pip
+  - passagemath-setup
+  - passagemath-environment
+  - cython
+  - cysignals
+  - numpy
+  - pkgconfig
+  - passagemath-abi == ${{ version }}
+  host:
+  - python
+  - sirocco
+  - numpy
+  - passagemath-modules
+  run:
+  - python
+  - cysignals
+  - passagemath-modules
+
+tests:
+- script: pytester
+  files:
+    recipe:
+    - test_passagemath_sirocco.py
+  requirements:
+    build:
+    - pytester
+    run:
+    - pytester-run
+    - passagemath-repl
+
+about:
+  license: GPL-2.0-or-later
+  license_file: README.rst
+  summary: 'passagemath: Certified root continuation with sirocco'
+  description: |
+    Distribution of a part of the Sage library.
+    It provides a Cython interface to the sirocco library for computing
+    topologically certified root continuation of bivariate polynomials.
+
+extra:
+  emscripten_tests:
+    python:
+      pytest_files:
+      - test_passagemath_sirocco.py
+  recipe-maintainers:
+  - mkoeppe

--- a/recipes/recipes_emscripten/passagemath-sirocco/recipe.yaml
+++ b/recipes/recipes_emscripten/passagemath-sirocco/recipe.yaml
@@ -30,6 +30,7 @@ requirements:
   - passagemath-abi == ${{ version }}
   host:
   - python
+  - gmp
   - sirocco
   - numpy
   - passagemath-modules

--- a/recipes/recipes_emscripten/passagemath-sirocco/test_passagemath_sirocco.py
+++ b/recipes/recipes_emscripten/passagemath-sirocco/test_passagemath_sirocco.py
@@ -1,0 +1,5 @@
+import pytest
+
+
+def test_import_passagemath_sirocco():
+    import passagemath_sirocco

